File: app/src/main/java/com/jeppsson/appexplore/db/PackageDatabase.java
package com.jeppsson.appexplore.db;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;

@Database(entities = {Package.class}, version = 1)
public abstract class PackageDatabase extends RoomDatabase {

    private static PackageDatabase INSTANCE;

    public abstract PackageDao dao();

    public static PackageDatabase getAppDatabase(Context context) {
        if (INSTANCE == null) {
            INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                    PackageDatabase.class, "package-db")
                    .build();
        }

        return INSTANCE;
    }
}

File: app/src/main/java/com/jeppsson/appexplore/db/PackageDatabase.kt
package com.jeppsson.appexplore.db

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import android.content.Context

@Database(entities = [Package::class], version = 1)
abstract class PackageDatabase : RoomDatabase() {

    abstract fun dao(): PackageDao

    companion object {

        private var INSTANCE: PackageDatabase? = null

        fun getAppDatabase(context: Context): PackageDatabase {
            val instance = INSTANCE
            if (instance != null) {
                return instance
            }

            val newInstance = Room.databaseBuilder(context.applicationContext,
                    PackageDatabase::class.java, "package-db")
                    .build()
            INSTANCE = newInstance

            return newInstance
        }
    }
}
